File: app/src/main/java/com/karlnosworthy/examples/daggerv2/ApplicationComponent.java
package com.karlnosworthy.examples.daggerv2;

import com.karlnosworthy.examples.daggerv2.activity.MainActivity;
import com.karlnosworthy.examples.daggerv2.activity.OtherActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    void inject(MainActivity mainActivity);

    void inject(OtherActivity otherActivity);
}

File: app/src/main/java/com/karlnosworthy/examples/daggerv2/ApplicationComponent.kt
package com.karlnosworthy.examples.daggerv2

import com.karlnosworthy.examples.daggerv2.activity.MainActivity
import com.karlnosworthy.examples.daggerv2.activity.OtherActivity

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {

    fun inject(mainActivity: MainActivity)

    fun inject(otherActivity: OtherActivity)
}
